Sandbox/BubbleSortLayer.cpp

```diff
@@ -0,0 +1,79 @@
+#include "BubbleSortLayer.h"
+
+#include <chrono>
+#include <thread>
+
+BubbleSortLayer::BubbleSortLayer()
+        : Aquarius::Layer("Bubble Sort ") {}
+
+void BubbleSortLayer::onCreation()
+{
+    Aquarius::Application* app = Aquarius::Application::get();
+    Aquarius::Window* window = app->getWindow();
+
+    int height = window->getHeight();
+    int width = window->getWidth();
+
+    m_Camera = std::make_shared<Aquarius::OrthographicCamera>(1, 0.01, height, width);
+
+    Aquarius::Renderer::Init();
+
+    // Populate the bar heights with random values
+    for (int i = 0; i < 100; i++)
+    {
+        barHeights[i] = rand() % 750 + 30;
+    }
+}
+
+void BubbleSortLayer::onUpdate(Aquarius::timeDelta_t ts)
+{
+    m_Camera->onUpdate(ts);
+    glm::vec3 cameraPos = m_Camera->getPosition();
+
+    Aquarius::Renderer::BeginScene(m_Camera.get());
+    Aquarius::Renderer::ClearColor({ 0.2, 0.3, 0.7 });
+    Aquarius::Renderer::Clear();
+
+
+    renderBars(100);
+
+    swapBars(i, j);
+    i = i+1;
+    j = j+1;
+
+    if(i == 99 and j ==100)
+    {
+        i = 0;
+        j = 1;
+    }
+
+    // This delay will allow for the changes occurring during the sort to be visible
+    std::this_thread::sleep_for(std::chrono::microseconds (500));
+}
+
+void BubbleSortLayer::renderBars(int size)
+{
+    int position = 0;
+
+    for(int i = 0; i < size; i++)
+    {
+        Aquarius::Renderer::DrawQuad(
+                {position, 0},
+                {10, barHeights[i]},
+                0,
+                {1,1,1,1}
+        );
```

Comment:
Nice

Reply:
would be cool to have color as a member in the layer so that we can easily change it

Reply:
Ya I can bring a member in for that, I tried it with random values using rand and it did not look good. You could not see the sort clearly.